### PR TITLE
#7108, #7119: Improve Geocarousel section 

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR improve the carousel sections of GeoStory by:

- refactoring the json structure to centralize the map configuration
- refactoring the carousel menu to align it to the navigation menu behaviours and avoid compressed cards
- fix bugs listed here https://github.com/geosolutions-it/MapStore2/issues/6993#issuecomment-880806131
- add icon for this type of section

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix

## Issue

**What is the current behavior?**

#7108
#7119

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

### CODE REVIEW

File: web/client/translations/data.en-US.json [hunk 1]
```diff
@@ -2438,6 +2438,9 @@
             "showFullscreenMap": "Show fullscreen map",
             "carouselAddTitle": "Add new card thumbnail",
             "carouselEditTitle": "Edit card thumbnail",
+            "carouselAddMapInfo": "Add a map background to configure position for carousel item",
+            "carouselAddItemInfo": "Associate current item to a position on the map to add more carousel item",
+            "carouselPlaceMarkerInfo": "Click anywhere on the map to place or reposition the marker",
```
Comment (allyoucanmap) [inline web/client/translations/data.en-US.json:2443]:
```suggestion
            "carouselAddMapInfo": "Add a map background",
            "carouselAddItemInfo": "Add a new marker location for the selected content",
            "carouselPlaceMarkerInfo": "Click on the map to add a new marker",
```

cc @tdipisa

File: web/client/translations/data.de-DE.json [hunk 2]
```diff
@@ -2426,6 +2426,9 @@
             "showFullscreenMap": "Vollbild-Anzeige Karte",
             "carouselAddTitle": "Neue Kartenminiaturansicht hinzufügen",
             "carouselEditTitle": "Kartenminiaturansicht bearbeiten",
+            "carouselAddMapInfo": "Fügen Sie einen Kartenhintergrund hinzu, um die Position für das Karussellelement zu konfigurieren",
+            "carouselAddItemInfo": "Verknüpfen Sie das aktuelle Element mit einer Position auf der Karte, um weitere Karussell-Elemente hinzuzufügen",
+            "carouselPlaceMarkerInfo": "Klicken Sie auf eine beliebige Stelle auf der Karte, um die Markierung zu platzieren oder neu zu positionieren",
```
Comment (allyoucanmap) [inline web/client/translations/data.de-DE.json:2431]:
```suggestion
            "carouselAddMapInfo": "Hinzufügen einer Hintergrundkarte",
            "carouselAddItemInfo": "Neue Markierung für das ausgewählte Karussellelement hinzufügen",
            "carouselPlaceMarkerInfo": "Klicken Sie auf die Karte, um eine neue Markierung hinzuzufügen",
```

File: web/client/translations/data.es-ES.json [hunk 3]
```diff
@@ -2426,6 +2426,9 @@
             "showFullscreenMap": "Ver mapa en pantalla completa",
             "carouselAddTitle": "Agregar nueva miniatura de tarjeta",
             "carouselEditTitle": "Editar miniatura de tarjeta",
+            "carouselAddMapInfo": "Agregue un fondo de mapa para configurar la posición del elemento del carrusel",
+            "carouselAddItemInfo": "Asociar el elemento actual a una posición en el mapa para agregar más elementos de carrusel",
+            "carouselPlaceMarkerInfo": "Haga clic en cualquier lugar del mapa para colocar o reposicionar el marcador",
```
Comment (allyoucanmap) [inline web/client/translations/data.es-ES.json:2431]:
```suggestion
            "carouselAddMapInfo": "Agregar un mapa de fondo",
            "carouselAddItemInfo": "Agregar un nuevo marcador para el elemento de carrusel seleccionado",
            "carouselPlaceMarkerInfo": "Haga clic en el mapa para agregar un nuevo marcador",
```

File: web/client/translations/data.fr-FR.json [hunk 4]
```diff
@@ -2427,6 +2427,9 @@
             "showFullscreenMap": "Montrer la carte en plein écran",
             "carouselAddTitle": "Ajouter une nouvelle vignette de carte",
             "carouselEditTitle": "Modifier la miniature de la carte",
+            "carouselAddMapInfo": "Ajouter un arrière-plan de carte pour configurer la position de l'élément du carrousel",
+            "carouselAddItemInfo": "Associez l'élément actuel à une position sur la carte pour ajouter plus d'éléments de carrousel",
+            "carouselPlaceMarkerInfo": "Cliquez n'importe où sur la carte pour placer ou repositionner le marqueur",
```
Comment (allyoucanmap) [inline web/client/translations/data.fr-FR.json:2432]:
```suggestion
            "carouselAddMapInfo": "Ajouter un fond de carte",
            "carouselAddItemInfo": "Ajouter un nouveau marqueur pour l'élément de carrousel sélectionné",
            "carouselPlaceMarkerInfo": "Cliquez sur la carte pour ajouter un nouveau marqueur",
```

File: web/client/translations/data.it-IT.json [hunk 5]
```diff
@@ -2427,6 +2427,9 @@
             "showFullscreenMap": "Visualizza mappa a pieno schermo",
             "carouselAddTitle": "Aggiungi nuova miniatura della carta",
             "carouselEditTitle": "Modifica la miniatura della carta",
+            "carouselAddMapInfo": "Aggiungi uno sfondo della mappa per configurare la posizione per l'elemento del carosello",
+            "carouselAddItemInfo": "Associa l'elemento corrente a una posizione sulla mappa per aggiungere più elementi del carosello",
+            "carouselPlaceMarkerInfo": "Fare clic in un punto qualsiasi della mappa per posizionare o riposizionare l'indicatore",
```
Comment (allyoucanmap) [inline web/client/translations/data.it-IT.json:2432]:
```suggestion
            "carouselAddMapInfo": "Aggiungi una mappa di sfondo",
            "carouselAddItemInfo": "Aggiungi un nuovo marker per l'elemento del carosello selezionato",
            "carouselPlaceMarkerInfo": "Clicca sulla mappa per aggiungere un nuovo marker",
```

**Review by dsuren1 — CHANGES_REQUESTED** [review]
Hi @allyoucanmap 
Changes with respect to draw support and background works fine. 
However I did see few change in behavior with respect to carousel card component update which needs to looked into and possibly needs a fix

- Clicking on carousel card again with map's zoom and position changed, doesn't reset to it's initial position when marker added (is this behavior replaced with map fit feature now ?) cc @tdipisa @allyoucanmap 
- Adding new item, the carousel widget is not scrolled to the added item
Scrollcontent
- Scroll event on carousel card item is propagated to other components
Mousescroll
- Also carousel item navigation with arrow keys is missing in the new geo carousel widget
- Infocard position is misaligned due to recent change

- After deleting an item in carousel, unable to select other items after the deleted item in list
deleteItem
- Selecting the card's footer toolbar icons, triggers card selection too (Is this behavior fine?  cc @tdipisa) 
autoSelection

**Comment by allyoucanmap** [comment]
Hi @dsuren1,
thanks for the review
done all fixes but these two:

> Clicking on carousel card again with map's zoom and position changed, doesn't reset to it's initial position when marker added (is this behavior replaced with map fit feature now ?) cc @tdipisa @allyoucanmap

yes, the new behaviour follow the fit feature workflow

> Also carousel item navigation with arrow keys is missing in the new geo carousel widget

The issue was referring to the arrows in the UI so for now we don't need an implementation for keyboard

**Review by dsuren1 — CHANGES_REQUESTED** [review]
> * Selecting the card's footer toolbar icons, triggers card selection too (Is this behavior fine?  cc @tdipisa)
Happens with delete icon (Edit doesn't trigger the selection )
- Popup support needs to be disabled when drawing, as it interferes with the draw displaying popup after placing the marker
- Small cosmetic, just so we know, in mobile view having two geocarousel section one after another, displays the second carousel widget partially in the bottom
2021-07-23 15_24_03-MapStore HomePage

Apart from these, the fixes works fine.

**Comment by allyoucanmap** [comment]
> Small cosmetic, just so we know, in mobile view having two geocarousel section one after another, displays the second carousel widget partially in the bottom

@dsuren1 done apart of this. Now the implementation uses position sticky so it's the default behaviour I would discuss before with @tdipisa but I don't think is a blocker. We will open a new issue if needed

**Comment by dsuren1** [comment]
LGTM :)

**Review by dsuren1 — APPROVED** [review]

**Pull request merged**